discovery/store.go

```diff
@@ -296,14 +299,18 @@ func (s *sqlStore) removeExpired() (int, error) {
 
 // updatePresentationRefreshTime creates/updates the next refresh time for a Verifiable Presentation on a Discovery Service.
 // If nextRegistration is nil, the entry will be removed from the database.
-func (s *sqlStore) updatePresentationRefreshTime(serviceID string, subjectID string, nextRefresh *time.Time) error {
+func (s *sqlStore) updatePresentationRefreshTime(serviceID string, subjectID string, parameters map[string]interface{}, nextRefresh *time.Time) error {
 	return s.db.Transaction(func(tx *gorm.DB) error {
 		if nextRefresh == nil {
 			// Delete registration
 			return tx.Delete(&presentationRefreshRecord{}, "service_id = ? AND subject_id = ?", serviceID, subjectID).Error
 		}
 		// Create or update it
-		return tx.Save(presentationRefreshRecord{SubjectID: subjectID, ServiceID: serviceID, NextRefresh: nextRefresh.Unix()}).Error
+		var bytes []byte
+		if parameters != nil {
+			bytes, _ = json.Marshal(parameters)
```

Comment:
returning err won't hurt?

Reply:
doesn't hurt but in general only possible when you put a function ptr in the map.